charts/kube-prometheus-stack/templates/alertmanager/networkpolicy.yaml

```diff
@@ -0,0 +1,79 @@
+{{- if and .Values.alertmanager.enabled .Values.alertmanager.networkPolicy.enabled }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "kube-prometheus-stack.fullname" . }}-alertmanager
+  namespace: {{ template "kube-prometheus-stack.namespace" . }}
+  labels:
+    app: {{ template "kube-prometheus-stack.name" . }}-alertmanager
+    {{- include "kube-prometheus-stack.labels" . | nindent 4 }}
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: alertmanager
+  policyTypes:
+    - Ingress
+    - Egress
+  ingress:
+    # Allow ingress from the ingress controller
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: {{ .Values.alertmanager.networkPolicy.ingressNS }}
+          podSelector:
+            matchLabels:
+              {{- toYaml .Values.alertmanager.networkPolicy.ingressPodLabels | nindent 14 }}
+      ports:
+        - port: {{ .Values.alertmanager.service.port }}
+          protocol: TCP
+        - port: {{ .Values.alertmanager.service.clusterPort }}
+          protocol: TCP
+    # Allow ingress from Prometheus
+    - from:
+        - podSelector:
+            matchLabels:
+              app.kubernetes.io/name: prometheus
+      ports:
+        - port: {{ .Values.alertmanager.service.port }}
+          protocol: TCP
+  egress:
```

Comment:
By default, egress should be not restricted. It can be an empty block from values and optionally filled via values.

For example, we have running a webhook pod inside the same namespace. Otherwise, end-users are using SMTP relay from a internal network. excluding internal network by default is not a good idea.

Reply:
Fixed by making egress optional and configurable through egress.enabled: false by default